packages/router/src/grpc/view-protocol-server/broadcast-transaction.ts

```diff
@@ -15,15 +19,38 @@ export const broadcastTransaction: Impl['broadcastTransaction'] = async (req, ct
   // start subscription early to prevent race condition
   const subscription = indexedDb.subscribe('TRANSACTION_INFO');
 
-  const id = await tendermint.broadcastTx(req.transaction);
+  const id = new TransactionId({ inner: await sha256Hash(req.transaction.toBinary()) });
```

Comment:
I like how you used the name `expectId` in `webapp/src/state/helpers`, and that could be nice here to make the purpose of this `id` clear. But up to you.